### PR TITLE
Apiv2 fixes

### PR DESCRIPTION
**Pull request by ffont**
Addresses two APIv2 issues:
* Issue in API root page due to wrong construction of dict
* Improves the way sound analysis data is passed to the sound serializers. This can potentially fix some memory leaks by now adding data directly to the view object

### CODE REVIEW

File: apiv2/serializers.py [hunk 1]
```diff
@@ -57,6 +57,12 @@ class AbstractSoundSerializer(serializers.HyperlinkedModelSerializer):
     default_fields = None
 
     def __init__(self, *args, **kwargs):
+        if 'sound_analysis_data' in kwargs:
+            # A sound_analysis_data is a dictionary with sound analysis informartion for sound ids
+            self.sound_analysis_data = kwargs.get('sound_analysis_data')
+            del kwargs['sound_analysis_data']
+        else:
+            self.sound_analysis_data = {}
```
Comment (Bomme) [inline apiv2/serializers.py:65]:
```suggestion
        self.sound_analysis_data = kwargs.pop('sound_analysis_data', {})
```

File: apiv2/apiv2_utils.py [hunk 2]
```diff
@@ -256,6 +257,18 @@ def initial(self, request, *args, **kwargs):
         self.get_request_information(request)
         self.store_monitor_usage()
 
+    def get_serializer(self, *args, **kwargs):
+        # Overwrite DRF's get_serializer method so that we add sound analysis information when requested
+        serializer_class = self.get_serializer_class()
+        kwargs['context'] = self.get_serializer_context()
+        if 'SoundListSerializer' in str(serializer_class):
+            # If we are trying to serialize sounds, check if we should and sound analysis data to them and add it
+            if type(args[0] == list):
```
Comment (Bomme) [inline apiv2/apiv2_utils.py:266]: `list` might be a bit restrictive. anything iterable should work.
maybe we can use 
`isinstance(args[0], Iterable)` instead?

**Review by Bomme — COMMENTED** [review]
I left some small comments :)

File: apiv2/apiv2_utils.py (hunk 2 above)
Comment (ffont) [reply apiv2/apiv2_utils.py:266]: Thanks, I'll update that :)

**Pull request merged**